File: library/src/main/java/com/github/dmitrikudrenko/countryphonecodechooser/utils/CountryCodeIntent.java
package com.github.dmitrikudrenko.countryphonecodechooser.utils;

import android.content.Intent;
import com.github.dmitrikudrenko.countryphonecodechooser.model.CountryCode;
import com.github.dmitrikudrenko.countryphonecodechooser.ui.CountryCodeChooserActivity;

public class CountryCodeIntent {
    public static CountryCode get(Intent intent) {
        if (intent != null) {
            return intent.getParcelableExtra(CountryCodeChooserActivity.EXTRA_COUNTRY);
        }
        return null;
    }
}

File: library/src/main/kotlin/com/github/dmitrikudrenko/countryphonecodechooser/utils/CountryCodeIntent.kt
package com.github.dmitrikudrenko.countryphonecodechooser.utils

import android.content.Intent
import com.github.dmitrikudrenko.countryphonecodechooser.model.CountryCode
import com.github.dmitrikudrenko.countryphonecodechooser.ui.CountryCodeChooserActivity

object CountryCodeIntent {
    operator fun get(intent: Intent?) =
            intent?.getParcelableExtra<CountryCode>(CountryCodeChooserActivity.EXTRA_COUNTRY)
}
